src/http.rs

```diff
@@ -507,14 +549,24 @@ pub type RawHeaderLine = (String, Vec<u8>);
 /// >                ; obsolete line folding
 /// >                ; see Section 3.2.4
 /// > ```
-pub fn read_header<R: Reader>(stream: &mut R) -> HttpResult<Option<RawHeaderLine>> {
+pub fn read_header<R: Read>(stream: &mut R) -> HttpResult<Option<RawHeaderLine>> {
+    macro_rules! byte (
```

Comment:
httparse is a push parser. My goal is to be like Pico, but using Iterator,
to keep things safe.

On Sun, Mar 1, 2015, 6:51 PM Jonathan Reem <notifications@github.com> wrote:

> In src/http.rs
> <https://github.com/hyperium/hyper/pull/354#discussion_r25576303>:
>
> > @@ -507,14 +549,24 @@ pub type RawHeaderLine = (String, Vec<u8>);
> >  /// >                ; obsolete line folding
> >  /// >                ; see Section 3.2.4
> >  /// > ```
> > -pub fn read_header<R: Reader>(stream: &mut R) -> HttpResult<Option<RawHeaderLine>> {
> > +pub fn read_header<R: Read>(stream: &mut R) -> HttpResult<Option<RawHeaderLine>> {
> > +    macro_rules! byte (
>
> Just found httpparse, seems interesting but we'll need a push parser for
> async integration later.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hyperium/hyper/pull/354/files#r25576303>.
>